.github/actions/upload/src/inputs.js

```diff
@@ -0,0 +1,161 @@
+import { resolve } from 'node:path'
+import { ethers } from 'ethers'
+import { ERROR_CODES, FilecoinPinError, getErrorMessage } from './errors.js'
+
+/**
+ * @typedef {import('./types.js').ParsedInputs} ParsedInputs
+ */
+
+/**
+ * Check if object has own property
+ * @param {any} object - Object to check
+ * @param {string} key - Property key
+ * @returns {boolean} True if object has own property
+ */
+const own = (object, key) => Object.hasOwn(object, key)
+
+/** @type {any} */
+let cachedInputsJson
+
+function readInputsJson() {
+  if (cachedInputsJson !== undefined) return cachedInputsJson
+
+  const raw = process.env.INPUTS_JSON
+  if (!raw) {
+    cachedInputsJson = null
+    return cachedInputsJson
+  }
+
+  try {
+    cachedInputsJson = JSON.parse(raw)
+  } catch (error) {
+    throw new Error(`Failed to parse INPUTS_JSON: ${error instanceof Error ? error.message : String(error)}`)
+  }
+
+  return cachedInputsJson
+}
+
+/**
+ * Convert value to string with fallback
+ * @param {any} value - Value to convert
+ * @param {string} fallback - Fallback value
+ * @returns {string} String representation
+ */
+function toStringValue(value, fallback = '') {
+  if (value === undefined || value === null) return String(fallback ?? '')
+  return typeof value === 'string' ? value : String(value)
+}
+
+/**
+ * Get input value from environment variables
+ * @param {string} name - Input name
+ * @param {string} fallback - Default value
+ * @returns {string} Input value
+ */
+export function getInput(name, fallback = '') {
+  const json = readInputsJson()
+  if (json && own(json, name)) {
+    return toStringValue(json[name], fallback).trim()
+  }
+
+  const envKey = `INPUT_${name.toUpperCase()}`
```

Comment:
so we don't need to replace capitals with `_` here? so `getInput('walletPrivateKey')` becomes `INPUT_WALLETPRIVATEKEY` and that works?